ray-operator/config/openshift/params.yaml

```diff
@@ -0,0 +1,7 @@
+varReference:
+  - path: subjects[]/namespace
+    kind: ClusterRoleBinding
+  - path: users[]
+    kind: SecurityContextConstraints
+  - path: spec/template/spec/containers[]/image
```

Comment:
I think the corresponding variable for parameterising the image is missing in `param.env`.

Reply:
Thanks @astefanutti  :)
I have address these changes in bda215576e97c9488c56e50bc1ae99fc61b2eddd

Reply:
I think we would need to have that variable actually used. We used to duplicate the entire manifests in the DW repository: https://github.com/opendatahub-io/distributed-workloads/tree/2049ceba5bd72efd6c49a189c6d0af2f0eca8a98/ray/operator

But now, I think we could only add a patch that overrides the KubeRay Deployment image in this overlay. We just need to make sure variables are correctly taken into account in patches.

Reply:
Hey @astefanutti 
Thank you for this. 

Just to clarify, would I be safe to remove params.yaml and params.env, as well as: 
```
configMapGenerator:
- name: ray-config
  envs:
    - params.env
configurations:
  - params.yaml

vars:
- name: namespace
  objref:
    kind: ConfigMap
    name: ray-config
    apiVersion: v1
  fieldref:
    fieldpath: data.namespace
```
Allowing us to set the image in kusomization.yaml and have the namspace injected by the ODH operator. Leaving us with:
    
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

namespace: opendatahub

resources:
- ray_operator_scc.yaml
- ../default

images:
- name: kuberay/operator
  newName: quay.io/kuberay/operator
  newTag: v0.6.0
```

```
kind: SecurityContextConstraints
apiVersion: security.openshift.io/v1
metadata:
  name: run-as-ray-user
seLinuxContext:
  type: MustRunAs
runAsUser:
  type: MustRunAs
  uid: 1000
users:
  - 'system:serviceaccount:$(namespace):kuberay-operator'

```

Reply:
@VanillaSpoon I don't think the namespace plugin will interpolate the `$(namespace)` variable in the SCC. But that has to be checked / tested.

For the image, my understand is that the ODH operator passes custom images as variables, not by setting `images` in the `kustomization.yaml` file.

I can see how your suggestion could simplify things, but I fear it won't be possible. Maybe we can keep the parameters approach in the short term, and reconsider it once we have that first working version?

Reply:
Thanks again @astefanutti 
I've pushed some updates to f9cbdefc4832dc3c3779dc53ad7282e43fb77f96 :)